packages/runtime-core/src/hydration.ts

```diff
@@ -938,10 +938,13 @@ function resolveCssVars(
   ) {
     const cssVars = instance.getCssVars()
     for (const key in cssVars) {
-      expectedMap.set(
-        `--${getEscapedCssVarName(key, false)}`,
-        String(cssVars[key]),
-      )
+      const value = cssVars[key]
+      if (isRenderableAttrValue(value)) {
```

Comment:
And I think it's not semantically correct to use `isRenderableAttrValue` here as the value isn't to be rendered as a DOM attribute. /cc @edison1105